floodviz/static/js/floodviz_onready.js

```diff
@@ -1,28 +1,48 @@
 document.addEventListener("DOMContentLoaded", function (event) {
-
 	"use strict";
 
-	// Map figure 
-	var map_options = {'height': FV.mapinfo.height,
-						'width': FV.mapinfo.width,
-						'proj': proj4(FV.mapinfo.proj4string), 
-						'bounds': FV.mapinfo.bounds,
-						'scale': FV.mapinfo.scale,
-						'bg_data': FV.mapinfo.bg_data,
-						'rivers_data': FV.mapinfo.rivers_data,
-						'ref_data': FV.mapinfo.ref_data, 
-						'site_data': FV.mapinfo.site_data,
-						'div_id': '#map'};
-	FV.map_figure = FV.mapmodule(map_options);
-	FV.map_figure.init();
+	// Map options
+	var map_options = {
+		'height': FV.mapinfo.height,
+		'width': FV.mapinfo.width,
+		'proj': proj4(FV.mapinfo.proj4string),
+		'bounds': FV.mapinfo.bounds,
+		'scale': FV.mapinfo.scale,
+		'bg_data': FV.mapinfo.bg_data,
+		'rivers_data': FV.mapinfo.rivers_data,
+		'ref_data': FV.mapinfo.ref_data,
+		'site_data': FV.mapinfo.site_data,
+		'div_id': '#map'
+	};
+	// Hydrograph options
+	var hydro_options = {
+		'height': FV.hydrograph_dimensions.height,
+		'width': FV.hydrograph_dimensions.width,
+		'div_id': '#hydrograph'
+	};
+
+	var map_frame = FV.mapmodule(map_options);
+	var hydro_frame = FV.hydromodule(hydro_options);
+
+
+	// Use frames to link interactions
+	hydro_options['site_tooltip_show'] = map_frame.site_tooltip_show;
```

Comment:
With further thought, I like my second option better where we pass the functions into the init. Probably still pass it in as an Object with properties rather than individual parameters.